app/models/post.rb

```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  def update_user_posts_counter
+    author.increment!(:posts_counter)
+    author.save
+  end
+
```

Comment:
- [ ] **[OPTIONAL]** There is no need to call `.save` here. We can also add a callback for this method and make it private so that when a post is created it is called automatically.

  ```
  after_save :update_user_posts_counter

  ...
  
  private

  def update_user_posts_counter
    author.increment!(:posts_counter)
  end
  ```